src/main/kotlin/org/jmailen/gradle/kotlinter/tasks/InstallPrePushHookTask.kt

```diff
@@ -0,0 +1,91 @@
+package org.jmailen.gradle.kotlinter.tasks
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.TaskAction
+import org.jmailen.gradle.kotlinter.support.findGitDir
+import org.jmailen.gradle.kotlinter.support.findGradlew
+import java.io.File
+
+/**
+ * Install or update the kotlinter-gradle pre-push hook.
+ */
+open class InstallPrePushHookTask : DefaultTask() {
+    @TaskAction
+    fun run() {
+        val dotGitDir = findGitDir(project.rootDir)
+        logger.info(".git directory: $dotGitDir")
+        val gradlew = findGradlew(project.rootDir).path
```

Comment:
I know it's crazy not to have a gradle wrapper installed, but what if we just abstracted this to `gradleCommand` and resolve it to `gradlew` or `gradlew.bat` when available, otherwise `gradle`? Do remember the extension on Windows!

Reply:
Ooh, great point about `…bat`.  I'll write something up that looks for `gradlew(.bat)?` and, if there isn't one, attempts to use a system-provided `gradle(.bat)?`.  I wish there were a way to just have Gradle tell you where it's executing from…

Reply:
Actually, I don't know how to make that work—we don't have a shell with which to find `gradle`, the `PATH` environment variable is around, so we could manually search that, but then I don't have a clue what to do on Windows.  Suggestions?

Reply:
Two possibilities.
1. We just do `gradle` and assume it's in their path, which we hope it is if they don't have wrapper installed.
2. We simply require gradle wrapper. If we don't find it, we exit with an error saying so.

Open to either. Which do you like?

Reply:
Oh, the first is a fine idea—I forgot that we don't actually need to interact with Gradle herein, just the script we're writing needs it at invocation time.  Easy, if the wrapper isn't found, I'll just use `gradle`.